### ISSUE TITLE
`text-encoding-identifier-case` error in JSX; `utf8` is not valid HTML

### ISSUE THREAD

**Issue by jp-knj** [sindresorhus/eslint-plugin-unicorn#1926]
According to [WhatWG](https://html.spec.whatwg.org/multipage/forms.html#attr-form-accept-charset), the value must be an [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) match for "UTF-8"

please text-encoding-identifier-case of rule check
Thanks for always helping us out

**example**
This is causing conflicts in our React code (in .tsx files that are linted by eslint):
```js
<form
  id="search_keyword"
  method="get"
  action={action}
  acceptCharset="UTF-8" // get eslint error
>
  <input
    type="search"
    name="q"
    id="q"
  />
        ....
</form>
```

**Comment by Pyrolistical** [comment]
Similarly `new TextDecoder("utf-8")` is another false positive as according to the [spec](https://encoding.spec.whatwg.org/#names-and-labels):
> Authors must use the [UTF-8](https://encoding.spec.whatwg.org/#utf-8) [encoding](https://encoding.spec.whatwg.org/#encoding) and must use its ([ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive)) "utf-8" [label](https://encoding.spec.whatwg.org/#label) to identify it.

`utf-8` is cannon.

**Comment by fisker** [comment]
It's hard to check places that only allows `utf-8`, but `utf-8` works everywhere (I think), how about we switch from `utf8` to use `utf-8`?

Mentioned this problem in orie propor, https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1690#issuecomment-1033685083

**Comment by timdavish** [comment]
> It's hard to check places that only allows `utf-8`, but `utf-8` works everywhere (I think), how about we switch from `utf8` to use `utf-8`?
> 
> Mentioned this problem in original proposal, [#1690 (comment)](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1690#issuecomment-1033685083)

Does it make sense to make this configurable, at a minimum? I know you originally proposed configurability [here](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1690#issue-1098573807)

**Comment by fisker** [comment]
Make sense to me.

**Comment by carlosjeurissen** [comment]
Situations exist in which utf8 is the default, and utf-8 is an alias like in nodeJS:
https://nodejs.org/api/buffer.html#buffers-and-character-encodings

However, considering utf-8 is required in more situations this seems the best option.

The following PR changes the default to utf-8: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2600

**Comment by carlosjeurissen** [comment]
@sindresorhus:
> The convention in Node.js (and as used in most Node.js docs) is utf8, and I don't see any reason to break that. I'm not against adding an option to support utf-8, but the default should continue to be utf8 (in our docs and code too).

Using this thread for further discussion. You feel specifically for nodeJS code (like fs.readFile), `utf8` should be preferred instead of `utf-8`? Or in general this eslint plugin should always default to `utf8` unless a user decides otherwise? (except for cases in which utf8 is not supported, like in https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2547)

**Comment by sindresorhus** [comment]
I prefer it to always default to `utf8` except the one case it cannot, unless the user has chosen to prefer `utf-8`.

**Comment by fisker** [comment]
```ts
{
	whatwgEncoding: boolean;
}
```

?

**Comment by clshortfuse** [comment]
"utf-8" is also the standard for Content-Type (eg: `application/json;charset=utf-8`). 

https://www.w3.org/International/articles/http-charset/index

https://www.iana.org/assignments/character-sets/character-sets.xhtml

It's a bit odd this eslint is asking to break with standards. "utf8" seems like a strictly NodeJS thing and is not in the standards.

**Comment by starjet** [comment]
This is such a questionable rule. Can this consistency even be maintained realistically? Some things may expect "utf-8" while some others may expect "UTF-8". I just turned the rule off and carried on, but I have to wonder why this rule is even there in the recommended rule set when it can easily break functionality
